### ISSUE TITLE
Commercial use of prebuilt images

### ISSUE THREAD

**Issue by Agalin** [CrunchyData/crunchy-containers#1430]
Source code for both the operator and all the images are available under Apache license. That's clear. But all the prebuilt binaries are hosted on CrunchyData Developer Portal which explicitly states everything provided there is not intended for production use without the support subscription.

Does it mean we should rebuild all the images to use them without a subscription? Docs state that CentOS-based (and now UBI) images are publicly available but if those cannot be used in production environments it's a bit misleading.

**Comment by jinnerbichler** [comment]
Good question! We are having the same issue.

**Comment by chadmayfield** [comment]
Very good question!  We are also having the same issue and would like further information on usage

**Comment by andrewlecuyer** [comment]
Thank you for reaching out!

This feedback is helpful.  We are in the process of updating our documentation/guidance in order to better clarify these details, but I wanted to go ahead and do so here as well.

As you correctly point out, the code within this repository (a collection of container image files associated with the containers that run in connection with [PGO](https://github.com/CrunchyData/postgres-operator)), and the PGO source code itself are both open source licensed under Apache. The various software components included within the Crunchy containers are also each open source and available subject to their applicable open source license (ex: Postgres, PostGIS, Patroni, etc.).

The Crunchy Container Suite and PGO pre-built images (binaries) are made available either through the [Crunchy Data Developer Portal](https://www.crunchydata.com/developers) (subject to the [Developer Program Terms](https://www.crunchydata.com/developers/terms-of-use) and not intended for production use) or to [Crunchy Data subscription](https://www.crunchydata.com/about/value-of-subscription/) customers.  Again, the feedback regarding the [documentation](https://access.crunchydata.com/documentation/postgres-operator/latest/) is helpful and we are working to update / clarify that language based on this feedback.

For use of Crunchy Data's pre-built binaries in commercial / production settings (outside of the uses contemplated by the Developer Program), we would hope that you would become a Crunchy Data subscription customer.  Crunchy Data sells this collection of binaries and support as the commercial offering [Crunchy Postgres for Kubernetes](https://www.crunchydata.com/products/crunchy-postgresql-for-kubernetes).

If it would be helpful to discuss your specific situation further, please reach out to our team at [info@crunchydata.com](mailto:info@crunchydata.com).

We appreciate the interest in PGO. Hopefully this is helpful clarification.

**Comment by Agalin** [comment]
Yeah, I think it makes it clear for us. Would be nice to include this info in README, similarly to how it's included in descriptions of older images on Docker Hub (e.g. https://hub.docker.com/r/crunchydata/crunchy-postgres)

**Comment by Agalin** [comment]
@andrewlecuyer continuing from https://github.com/CrunchyData/postgres-operator/issues/3218 as it's related. What about the new `postgres-operator-upgrade` image introduced in PGO 5.1.0? Is it supposed to be a closed-source component? I don't see it neither here nor in postgres-operator repo, not even in Crunchy Data developer portal.

**Comment by szelenka** [comment]
@andrewlecuyer can you comment if `postgres-operator-upgrade` (i.e. registry.developers.crunchydata.com/crunchydata/postgres-operator-upgrade:ubi8-5.1.3-0) is a closed-source image that's only available with a valid subscription license?

Or where can we locate the source code for this image if it is licensed under Apache License 2.0?

**Comment by Agalin** [comment]
Looks like it's finally there. Today's commit to postgres-operator repo adds upgrade logic.

**Comment by shinyaaa** [comment]
@Agalin 
>Looks like it's finally there. Today's commit to postgres-operator repo adds upgrade logic.

Are you referring to [this commit](https://github.com/CrunchyData/postgres-operator/commit/0569dfcee312c8cecbe1c5cbd38d500d21699208)?
I don't think this commit provides a Dockerfile to build the `postgres-operator-upgrade` image.

**Comment by Agalin** [comment]
Yes, it's about this commit. It indeed doesn't provide a dockerfile but to my understanding there won't be a separate container anymore - this commit introduces upgrade handling logic to the main operator app.
